packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -412,19 +419,35 @@ export async function transformGlobImport(
 
           const resolvePaths = (file: string) => {
             if (!dir) {
```

Comment:
This specific `dir` like my other comment may need to be updated too. In the code above it leads to:

https://github.com/vitejs/vite/blob/ce0eec91c9b5a91c52bf5b9bae6173d35f6e1649/packages/vite/src/node/plugins/importMetaGlob.ts#L376-L383

It should also consider `base` like shown in my comment. However, we don't know about `base` until `parseImportGlob()` is called.

Maybe a refactor here could be to remove this `dir` entirely. `parseImportGlob` can return and we access from here instead:

https://github.com/vitejs/vite/blob/ce0eec91c9b5a91c52bf5b9bae6173d35f6e1649/packages/vite/src/node/plugins/importMetaGlob.ts#L392-L393

e.g.

```js
async ({ globsResolved, isRelative, options, index, start, end, dir }) => {
```

Since we only need the `dir` within that callback. Free to explore other options to refactor this too though.

Reply:
It would be great to make the `dir` reusable. I'll try refactoring it.